### ISSUE TITLE
Order dependent `override` error when subclassing with multiple typevars with defaults

### ISSUE THREAD

**Issue by randolf-scholz** [python/mypy#19362]
**Bug Report**

This false positive `override` error seems to happen when multiple typevars with defaults are present, and we are using them in a different order than the original one.

**To Reproduce**

Code sample in [pyright playground](https://pyright-play.net/?code=MYGwhgzhAEAqBOYB2EBmB7eBbA2gDWgF5p0AjAKwFNgAXAGmgE0iSLqaBdALgChp-oAE0qpowdAAcAngAoIlEKgCU0ALQA%2BaACIEyNJlx4GjDlq7QAdFb4DhogJZIAbpXg05C5Ws07EKDNg4jAx4puZWFjxRoJAwAGLo6PgsZFS0xilstBwyuv4G%2BMYcSrwCQiJikrLyiioa2glJRkxhllb8AMTQWFLS5oCg5NDS8PYA5gAWNAM2-HbQji5uHrXeDYlBIa0Rnd29UuaAMuRDUiMTU9D9PEA)

```python
class Transform[X = object, Y = object]:
    def copy(self) -> "Transform[X, Y]": ...
    def invert(self) -> "Transform[Y, X]": ...


class Foo[X = object, Y = object](Transform[X, Y]):
    def copy(self) -> "Foo[X, Y]": ...  # mypy: ✅ pyright: ✅
    def invert(self) -> "Foo[Y, X]": ...  # mypy: ❌ pyright: ✅
```

```
Return type "Foo[Y, X]" of "invert" incompatible with return type "Transform[Y, X]" in supertype "Transform"  [override]
```

with legacy syntax: <https://mypy-play.net/?mypy=latest&python=3.12&gist=df2aabb39005153868c4bcb7d880e921>, Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgGoCGIANFAOLErEhIDGAUGwBpQC8hJFEAAoARJxE0AJsWDkArgBsYPMACMAVsRYwAlGwCavfgjKVR%2BiVGmzFytZu16AqkaInBop5evylKjVq6bKSuAmYipN4yvnYBjhwsCuQAzsmEIOQoycC4EEL0jMwsANqcNPoAujoAXGxQ9VYyUCxgCHBCycQKwDpQALQAfFAiBBlZOSAQpeUVItVQAHRLdQ3WmCgAbkwwHV09-UMjY9m5xfo0nLPzSwsJSalQAGJgYEKjmSeTxU40pBU0BSYrG%2BvyqtQajWAzVa7U63V6g2GzzAIKgfzmiyW9QAxFAIIg4PNAKDkUDazDQAAsYMSVvU1qgtnhdvCDkiXsVSDQnFdMQscXiCfNADLkpLg5KpNKAA)

```python
from typing import TypeVar, Generic

X = TypeVar("X", default=object)
Y = TypeVar("Y", default=object)
U = TypeVar("U", default=object)
V = TypeVar("V", default=object)

class Transform(Generic[X, Y]):
    def copy(self) -> "Transform[X, Y]": ...
    def invert(self) -> "Transform[Y, X]": ...

class Foo(Transform[U, V], Generic[U, V]):
    def copy(self) -> "Foo[U, V]": ...  # mypy: ✅ pyright: ✅
    def invert(self) -> "Foo[V, U]": ...  # mypy: ❌ pyright: ✅
```
